WalletWasabi/Blockchain/TransactionOutputs/SmartCoin.cs

```diff
@@ -167,6 +168,23 @@ public bool RefreshAndGetIsBanned()
 
 	public bool IsReplaceable() => Transaction.IsRBF;
 
+	/// <returns>False if external, or the tx inputs are all external.</returns>
+	/// <remarks>Context: https://github.com/zkSNACKs/WalletWasabi/issues/10567</remarks>
+	public bool IsSufficientlyDistancedFromExternalKeys()
```

Comment:
@Kruwed you might want to look at this code to ensure we didn't misunderstood each other.

Reply:
I don't know much programming, but I am going to test it today to see if it works as expected.

Reply:
Scenarios I tested:
- Round created with only nonprivate coins created semi-private coins
- Round created with private + nonprivate coins created private coins
- Round created with semi private coins + nonprivate coins created private coins
- Round created with private coins + semi-private coins created private coins

^ All of this is correct behavior for https://github.com/zkSNACKs/WalletWasabi/issues/10567

For comparison, https://github.com/zkSNACKs/WalletWasabi/issues/10517 would consider the outputs from the top 3 scenarios semi-private and only consider the bottom one to create private coins.